v0/destinations/pardot/util.js

```diff
@@ -0,0 +1,170 @@
+        stage,
+        scope: TRANSFORMER_METRIC.MEASUREMENT_TYPE.API.SCOPE,
+        meta: TRANSFORMER_METRIC.MEASUREMENT_TYPE.API.META.RETRYABLE
+      }
+    };
+  }
+  return {
+    status: 400,
+    stats: {
+      destination: DESTINATION,
+      stage,
+      scope: TRANSFORMER_METRIC.MEASUREMENT_TYPE.API.SCOPE,
+      meta: TRANSFORMER_METRIC.MEASUREMENT_TYPE.API.META.ABORTABLE
+    }
+  };
+};
+
+const pardotRespHandler = (destResponse, stageMsg, stage) => {
+  const { status, response } = destResponse;
+  const respAttributes = response["@attributes"];
+  const { stat, err_code: errorCode } = respAttributes;
+
+  if (isHttpStatusSuccess(status) && stat !== "fail") {
+    // Mostly any error will not have a status of 2xx
+    return response;
+  }
+  const statusAndStats = getStatusAndStats(errorCode, stage);
+  throw new ErrorBuilder()
+    .setStatus(statusAndStats.status)
+    .setDestinationResponse(response)
+    .setMessage(`Pardot: ${response.err} ${stageMsg}`)
+    .setAuthErrorCategory(getAuthErrCategory(errorCode))
+    .setStatTags(statusAndStats.stats)
+    .build();
+};
+
+const responseHandler = (destinationResponse, _dest) => {
+  const message = `[Pardot Response Handler] - Request Processed Successfully`;
+  const { status } = destinationResponse;
+  // else successfully return status, message and original destination response
+  pardotRespHandler(
+    destinationResponse,
+    "during Pardot response transformation",
+    TRANSFORMER_METRIC.TRANSFORMER_STAGE.RESPONSE_TRANSFORM
+  );
+  return {
+    status,
+    message,
+    destinationResponse
+  };
+};
+
+/**
+ * depricating: handles proxying requests to destinations from server, expects requsts in "defaultRequestConfig"
+ * note: needed for test api
+ * @param {*} request
+ * @returns
+ */
+const pardotProxyRequest = async request => {
+  const { body, method, params, endpoint } = request;
+  let { headers } = request;
+  let data;
+  let payload;
+  let payloadFormat;
+  for (const [key, value] of Object.entries(body)) {
+    if (!isEmpty(value)) {
+      payload = value;
+      payloadFormat = key;
+    }
+  }
+
+  switch (payloadFormat) {
+    case "JSON_ARRAY":
+      data = payload.batch;
+      // TODO: add headers
+      break;
+    case "JSON":
+      data = payload;
+      headers = { ...headers, "Content-Type": "application/json" };
+      break;
+    case "XML":
+      data = `${payload}`;
+      headers = { ...headers, "Content-Type": "application/xml" };
+      break;
+    case "FORM":
+      data = new URLSearchParams();
+      data.append("format", "json");
+      Object.keys(payload).forEach(key => {
+        data.append(`${key}`, `${payload[key]}`);
+      });
+      headers = {
+        ...headers,
+        "Content-Type": "application/x-www-form-urlencoded"
+      };
+      break;
+    case "MULTIPART-FORM":
+      // TODO:
+      break;
+    default:
+      console.debug(`body format ${payloadFormat} not supported`);
```

Comment:
throw error here?

Reply:
It was changed from error throwing to the current way. Not sure about the reason behind it.